pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/BasePartitionUpsertMetadataManager.java

```diff
@@ -514,6 +521,24 @@ public void removeSegment(IndexSegment segment) {
         return;
       }
     }
+
+    if (_helixManager != null) {
+      try {
+        HelixDataAccessor dataAccessor = _helixManager.getHelixDataAccessor();
+        PropertyKey propertyKey = dataAccessor.keyBuilder().idealStates(_tableNameWithType);
+        IdealState idealState = dataAccessor.getProperty(propertyKey);
+        if (!idealState.isEnabled()) {
```

Comment:
For my learning, what’s the effect to disable a table? Because I’d wonder if we could reuse some of the logic of dropping table on the server (as that logic was improved to avoid cleaning up upsert states for each individual segment).

Reply:
As discussed offline, that is not possible because disable flow only sets a flag in ideal state and then helix takes over removing the segments

in drop flow, we have the control and actually pass a custom message to all servers which then take over segment removal

we also can't stop table upsert metadata manager cause a disabled table can be re-enabled as well